common-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java

```diff
@@ -41,19 +41,43 @@ public ConceptService(EntityManager entityManager) {
   public Slice<Concept> searchConcepts(String query,
       StandardConceptFilter standardConceptFilter, List<String> vocabularyIds,
       List<String> domainIds, int limit) {
+
+      //String[] keywords=query.split("\\s*(=>|,|\\s)\\s*");
+      String[] keywords=query.split("[,+\\s+]");
+      for(int i=0;i<keywords.length;i++){
+          String key=keywords[i];
+          if(key.length() < 3){
+              key="\""+key+"\"";
+              keywords[i]=key;
+          }
+      }
+
+      String query2=new String();
+      for(String key:keywords){
+          if(query2.length()==0){
+              query2="+"+key;
+          }else if(key.contains("\"")){
+              query2=query2+key;
+          }else{
+              query2=query2+"+"+key;
+          }
+      }
+
+      final String keyword=query2;
+
     Specification<Concept> conceptSpecification =
         (root, criteriaQuery, criteriaBuilder) -> {
             List<Predicate> predicates = new ArrayList<>();
 
             // Check that the concept name, code, or ID matches the query string.
             List<Predicate> queryPredicates = new ArrayList<>();
             Expression<Double> matchExp = criteriaBuilder.function("match", Double.class,
-                root.get("conceptName"), criteriaBuilder.literal(query));
+                root.get("conceptName"), criteriaBuilder.literal(keyword));
             queryPredicates.add(criteriaBuilder.greaterThan(matchExp, 0.0));
             queryPredicates.add(criteriaBuilder.equal(root.get("conceptCode"),
-                criteriaBuilder.literal(query)));
+                criteriaBuilder.literal(keyword)));
```

Comment:
This is doing an exact match on concept code, not a MATCH. (Given how concept codes are, I don't think it makes sense to do MATCH logic on them.) So you should be using query here, not keyword.

Reply:
Got it.